src/headwordsmodel.cc

```diff
@@ -77,36 +78,28 @@ void HeadwordListModel::appendWord( const QString & word )
   words.append( word );
 }
 
-void HeadwordListModel::requestFinished()
+void HeadwordListModel::requestFinished( const sptr< Dictionary::WordSearchRequest > & request )
```

Comment:
I don't have time to check if there is anything wrong.

But, shared_ptr increases ref count when copying. Passing shared_ptr by reference feels wrong 😅

Reply:
`connect( sr.get(), &Dictionary::Request::finished, this, [ this, sr ]() {  `

the lambda capture the sr by copying , so it should be ok?